Mk/lex-rules.am

```diff
@@ -1,6 +1,6 @@
 %.y: %.ym $(syslog_ng_tools)/merge-grammar.py $(syslog_ng_tools)/cfg-grammar.y
 	$(AM_V_at) $(mkinstalldirs) $(dir $@)
-	$(AM_V_GEN) $(syslog_ng_tools)/merge-grammar.py $< > $@
+	$(AM_V_GEN) $(PYTHON) $(syslog_ng_tools)/merge-grammar.py $< > $@
```

Comment:
The CMake also has to use PYTHO for the grammar merge. Hint: `cmake/Modules/GenerateYFromYm.cmake`.

Reply:
Thanks, I missed that. Sorry!

Reply:
Fixed (I hope).

Reply:
Maybe my comment was misleading the PYTHON variable in cmake does not exists, and does not point to the python interpreter. The `PYTHON_EXECUTABLE` does.

But still there is also a reason why cmake version won't work as intended.

The `with-python` and the option in cmake was not intended to specify which python version you are using for build, but rather for the module.

In cmake the python detection was "hidden" from the upper level, and determined in the python module. To follow up the same change in cmake as in autotools, one probably should move the python detection to the upper level in cmake.

Reply:
Sure, I hope on next Monday I am going to be in the office, and we can check it together.